src/huggingface_hub/repocard_data.py

```diff
@@ -184,6 +184,8 @@ def to_dict(self) -> Dict[str, Any]:
         """
 
         data_dict = copy.deepcopy(self.__dict__)
+        if original_order:
+            data_dict = {k: data_dict[k] for k in original_order + list(set(data_dict.keys()) - set(original_order))}
```

Comment:
Thanks for the prompt changes @hlky ! 

As a nit before merging, I would move this logic to the `to_yaml` part so that everything related to formatting is kept in a single place. Even though python dicts are ordered by default, it's still "weird" to do `object.to_dict(..., original_order=...)` since the return value is still a dict.